### ISSUE TITLE
[exporter/otelarrow] rename config options `disabled` and `disable_downgrade` to prevent double negation

### ISSUE THREAD

**Issue by andrzej-stencel** [open-telemetry/opentelemetry-collector-contrib#33008]
### Component(s)

exporter/otelarrow

### Describe the issue you're reporting

The Otel-arrow exporter defines a configuration option named `disabled` in the `arrow` section of the configuration: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/c5485bf4ddef9c0acce85bff1fed60f669aabf6a/exporter/otelarrowexporter/config.go#L72

This option should rather be named `enabled` to prevent double negation mind-benders like `if !c.disabled`.

Same with `disable_downgrade`: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/c5485bf4ddef9c0acce85bff1fed60f669aabf6a/exporter/otelarrowexporter/config.go#L77

I think `downgrade_enabled` is a better name.

**Comment by crobert-1** [comment]
I agree that `enabled` is more clear than `disabled`, and would add that `enabled` is usually what the collector uses for options like this. I'll defer to code owners for more feedback here. This would be a breaking change.

_Removing `needs triage`._

**Comment by jmacd** [comment]
Will do.

**Comment by jmacd** [comment]
I feel it's worth challenging this request. The normal operating mode of the OTel-Arrow components is to support both OTLP and OTAP. When the user wants to change from the norm, they disable something. When we have an optional field, the zero value should be the default state, which is enabled. To have a zero-value mean disabled, we need a variable named "disabled".

I don't see why double-negation in code matters, e.g., `if !c.disabled`, as much as the perception that a user can disable something if they don't want the default behavior.
=
